server/service/hosts.go

```diff
@@ -631,6 +631,54 @@ func processHostFilters(ctx context.Context, opt fleet.HostListOptions, lid *uin
 	return filter, nil
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// List Host Device Mappings
+////////////////////////////////////////////////////////////////////////////////
+
+type listHostDeviceMappingRequest struct {
+	ID uint `url:"id"`
+}
+
+type listHostDeviceMappingResponse struct {
+	DeviceMapping []*fleet.HostDeviceMapping `json:"device_mapping"`
+	Err           error                      `json:"error,omitempty"`
+}
+
+func (r listHostDeviceMappingResponse) error() error { return r.Err }
+
+func listHostDeviceMappingEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (interface{}, error) {
+	req := request.(*listHostDeviceMappingRequest)
+	dms, err := svc.ListHostDeviceMapping(ctx, req.ID)
+	if err != nil {
+		return listHostDeviceMappingResponse{Err: err}, nil
+	}
+	return listHostDeviceMappingResponse{DeviceMapping: dms}, nil
+}
+
+func (svc *Service) ListHostDeviceMapping(ctx context.Context, id uint) ([]*fleet.HostDeviceMapping, error) {
+	if err := svc.authz.Authorize(ctx, &fleet.Host{}, fleet.ActionList); err != nil {
+		return nil, err
+	}
+
+	// TODO(mna): this is a pattern that is used elsewhere for hosts, authorize
+	// on list, then load the host to get the team info, and authorize properly
+	// (read, with team_id filled). I wonder if we should add a "quick load" of
+	// host when used just for that purpose, because loading even without the
+	// extra info is still a big-ish query with potentially lots of columns and
+	// at least 4 tables involved.
```

Comment:
Good idea! (`Datastore.HostForAuth`?)

Reply:
Created a ticket to track this: https://github.com/fleetdm/fleet/issues/3439